### ISSUE TITLE
feature request(cli): option to pass file containing filenames to lint

### ISSUE THREAD

**Issue by vegerot** [quick-lint/quick-lint-js#1166]
Problem:
I want to run quick-lint on a large project (120k files, 6m SLoC).

Approaches tried
- `quick-lint-js $(fd ...)` zsh can't handle that many arguments
- manually calling `exec("/usr/local/bin/quick-lint-js", ...)` fails because it exceeds `NCARGS`
- writing my own script that batches calls to quick-lint in batch sizes of `NCARGS` works but is inconvenient to users

Recommendation:
quick-lint CLI option that lets accepts the path to a file containing the paths of many files to check.  This file can contain many more files than `NCARGS`.

```sh
$ fd --extension=.js --extension=.mjs --extension=.jsx --extension=.ts --extension=.tsx > allfiles.txt
$ quick-lint-js --file-with-filenames=allfiles.txt
```

Out of scope:
If we're going to be linting hundreds of thousands of files anyways, we might as well do some of them in parallel.  Surely once we're >100k files the overhead of parallelism is worth it.

**Comment by vegerot** [comment]
note: #1167 would be an alternative solution if quick-lint respected `.gitignore`.  However, since that proposal does not, this proposal is better.

**Comment by vegerot** [comment]
I accept this task.

**Comment by strager** [comment]
Why not use `xargs`?

**Comment by vegerot** [comment]
So something like 

```sh
$ fd --extension=.js --extension=.jsx --extension=.ts --extension=.tsx | xargs -n999999999 quick-lint-js
```
?  Because that makes too much sense.

**Comment by vegerot** [comment]
think of the microseconds saved from quick-lint being able to do it all itself!

**Comment by vegerot** [comment]
`xargs` was the first thing I looked into when I played with this idea this week (after `fd`'s builtin exec).  The problem I ran into was not being able to tell `xargs` to wrap quotes around every argument

```sh
❯ echo "apps/us/e2e/full-regression/src/glass-orders/tc196-tiered -fast-refund -order.test.ts" | xargs -J{} quick-lint-js {}
error: unrecognized option: -fast-refund
```

That's when I tried the other solutions I mentioned in the OP and forgot I had tried this!  My bad

**Comment by strager** [comment]
@vegerot You should look at the `-0` option of both `fd` and `xargs`. `-0` on both will cause file names to be transferred without space splitting.

**Comment by strager** [comment]
Typical command: `find . -name '*.ts' -print0 | xargs -0 quick-lint-js --`
